eng/ci/public-build.yml

```diff
@@ -0,0 +1,38 @@
+name: $(Build.SourceBranchName)_$(Build.Reason)
+
+pr:
+  branches:
+    include:
+    - v4.x
+    - release_4.0
+    - v3.x
+    - release_3.0
+
+trigger:
+  batch: true
+  branches:
+    include:
+    - v4.x
+    - release_4.0
+    - v3.x
+    - release_3.0
+
+resources:
+  repositories:
+  - repository: 1es
+    type: git
+    name: 1ESPipelineTemplates/1ESPipelineTemplates
+    ref: refs/tags/release
+
+extends:
+  template: v1/1ES.Unofficial.PipelineTemplate.yml@1es
+  parameters:
+    pool:
+      name: 1es-pool-azfunc-public
+      image: 1es-windows-2022
+      os: windows
+    stages:     
+      - stage: BuildAndTest
+        jobs:
+        - template: /eng/ci/templates/official/jobs/build-test-public.yml@self
```

Comment:
The codeQL file needs to be removed and you need to explicitly tell the public-build.yml to run it.